schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/dao/common/ActivityPackageActivityDao.java

```diff
@@ -60,4 +153,40 @@ public ByOrganizationAndActivityQuery(ActivityPackageActivityMeta meta, Key orga
                     .asKeyList();
         }
     }
+
+    private static class ByActivityPackageAndActivityQuery extends BaseDaoQuery<ActivityPackageActivity, ActivityPackageActivityMeta> {
+        public ByActivityPackageAndActivityQuery(ActivityPackageActivityMeta meta, Key organizationId, Key activityPackageId, Key activityId) {
+            super(meta, new Serializable[]{organizationId, activityPackageId, activityId});
+        }
+
+        @Override
+        public List<Key> execute() {
+            Key organizationId = parameters.get(0);
+            Key activityPackageId = parameters.get(1);
+            Key activityId = parameters.get(2);
+            return Datastore
+                    .query(Datastore.getCurrentTransaction(), meta, organizationId)
+                    .filter(new CompositeCriterion(meta,
+                            Query.CompositeFilterOperator.AND,
+                            meta.activityPackageRef.equal(activityPackageId),
+                            meta.activityRef.equal(activityId)))
+                    .asKeyList();
```

Comment:
Not really an issue, but `.filter(Crit... crits)` method is an implicit AND (I might be wrong, but I'm sure we would catch it with a test).  But what it means is, you could say
```java
.filter (meta.actPackRef.equals(aId), meta.actRef.equal(aId))
```